staging/src/k8s.io/apimachinery/pkg/api/meta/testrestmapper/test_restmapper.go

```diff
@@ -93,21 +93,14 @@ func prioritiesForGroups(m *registered.APIRegistrationManager, groups ...string)
 }
 
 func newRESTMapper(scheme *runtime.Scheme, groupMeta *apimachinery.GroupMeta) meta.RESTMapper {
-	// the list of kinds that are scoped at the root of the api hierarchy
-	// if a kind is not enumerated here, it is assumed to have a namespace scope
-	rootScoped := sets.NewString()
-	if groupMeta.RootScopedKinds != nil {
-		rootScoped = groupMeta.RootScopedKinds
-	}
-
 	mapper := meta.NewDefaultRESTMapper(groupMeta.GroupVersions)
 	for _, gv := range groupMeta.GroupVersions {
 		for kind := range scheme.KnownTypes(gv) {
 			if ignoredKinds.Has(kind) {
 				continue
 			}
 			scope := meta.RESTScopeNamespace
-			if rootScoped.Has(kind) {
+			if rootScopedKinds[gv.WithKind(kind).GroupKind()] {
```

Comment:
this is going to drift and get weird quick...

Reply:
> this is going to drift and get weird quick...

It's for tests.  Those tests should really be using a very tight version of this.  They won't be affected by changes to the actual objects.  This isn't used to make contact to a real server.